utilities/dataflow_pubsub_to_hcapi/dataflow_pubsub_to_hcapi.py

```diff
@@ -0,0 +1,461 @@
+                "Authorization": f"Bearer {authToken}",
+                "Prefer": "handling=strict"
+            }
+            logging.info("Posting FHIR resource to HCAPI")
+            payload_data =json.loads(data)
+            reqSession = self.createRequestSession()
+            response = requests.post(url, headers=authHeaders, json=payload_data,timeout=30)
+            response.raise_for_status()
+            responseJSON = response.json()
+        except Exception as error:
+            reqSession.close()
+            logging.error("process to POST FHIR resource failed")
+            errorMessage = "Process to POST FHIR Resource to HCAPI failed due to : {}".format(str(type(error).__name__)+" --> "+ str(error))
+            responseJSON = dict()
+            responseJSON['errorMessage'] = errorMessage
+            responseFlag = "fail"
+            return responseJSON,responseFlag
+        else:
+            reqSession.close()
+            return responseJSON,responseFlag
+
+    def post_fhir_message(self, resourceType:str,payload:str):
+        """ Function to post messages to FHIR store """
+        """
+        :param str resourceType: FHIR Resource to post data
+        :param str payload: FHIR Resource as string
+        :return dict responseJSON: response in JSON based on criteria
+        :return str responseFlag: repsonse status pass or fail
+        """
+        resourceType=str(resourceType)
+        url = self.hcapi_fhir_url(store=self.hcapi_post_fhir_store)+f'/{resourceType}'
+        postResponse,responseFlag = self.hcapi_post(url,payload)
+        return postResponse,responseFlag
+
+    def get_fhir_message(self, url_path:str):
+        """ Function to fetch FHIR resources from HCAPI"""
+        """
+        :param str url_path: FHIR resource URL to get FHIR resource from HCAPI
+        :return dict responseJSON: response in JSON based on criteria
+        :return str responseFlag: repsonse status pass or fail
+        """
+        url_path=str(url_path)
+        url = self.hcapi_fhir_url(store=self.hcapi_get_fhir_store)+f'/{url_path}'
+        responseJSON, responseFlag = self.hcapi_get(url)
+        return responseJSON, responseFlag
+
+
+class transformations:
+    """Generic class to build multiple FHIR resource as per business requirements.
+        For example, we are building Patient FHIR resource """
+    
+    def buildIdentifier(message:dict, messageKey:str):
+        """Function to build mulitple identifiers for a given FHIR resource"""
+        """
+        :param dict message: raw message
+        :param str messageKey: identifier value example SSN, ID, etc.
+        :return dict identifier: Identifier dictionary with system and value as per US Core
+        """
+        identifier= dict()
+        identifier['system'] = str(messageKey)
+        identifier['value'] = message[messageKey] if message[messageKey] else "N/A"
+        return identifier
+
+    @staticmethod
+    def buildPatientResource(message:dict):
```

Comment:
we can remove this function right? Isn't this specific to the CSV info coming in?

Reply:
No this is specific for JSON payload.